File: February/feb14/feb14.java
class Solution {
    private TreeNode first = null, second = null, prev = null;

    // Inorder traversal to find misplaced nodes
    private void inorder(TreeNode root) {
        if (root == null) return;

        inorder(root.left); // Traverse left subtree

        // Identify swapped nodes
        if (prev != null && root.val < prev.val) { 
            if (first == null) first = prev; // First incorrect node
            second = root;  // Second incorrect node (updated if needed)
        }
        
        prev = root; // Move to the next node

        inorder(root.right); // Traverse right subtree
    }

    public void correctBST(TreeNode root) {
        inorder(root); // Identify misplaced nodes
        if (first != null && second != null) {
            int temp = first.val;
            first.val = second.val;
            second.val = temp; // Swap values to restore BST
        }
    }
}

File: February/feb14/feb14.py
class Solution:
    def __init__(self):
        self.first = self.second = self.prev = None

    def inorder(self, root):
        if not root:
            return

        self.inorder(root.left)  # Traverse left subtree

        # Identify swapped nodes
        if self.prev and root.val < self.prev.val:
            if not self.first:
                self.first = self.prev  # First incorrect node
            self.second = root  # Second incorrect node (updated if needed)

        self.prev = root  # Move to the next node

        self.inorder(root.right)  # Traverse right subtree

    def correctBST(self, root):
        self.inorder(root)  # Identify misplaced nodes
        if self.first and self.second:
            self.first.val, self.second.val = self.second.val, self.first.val  # Swap values to restore BST
